### ISSUE TITLE
Downloading results on 1.49 putting double quotes around numbers that are at least 4 digits long, and changes Date format

### ISSUE THREAD

**Issue by FilmonK** [metabase/metabase#40482]
### Describe the bug

We have a customer who's downloading results to CSV and XLXS, and noticed number data types with 4 digits or more are being placed within double quotes.

"2,450"
vs
872

The date format has also changed on exports where on previous versions such as 1.48.6 it would be **2022-03-30**, and 1.49 it's **“March 30, 2022”**.
The Localization settings have not been changed between versions.

I'm using a postgres instance and able to replicate it on 1.49 but just with CSV exports.
Their backend is Athena, on 1.49.1 and it's happening with both exports. I haven't had a chance to test simulating their configuration.

### To Reproduce

Using a postgres db on version 15.3 for these tests.

1. Have a table with a numeric based column. For this test I'm using Integer and Decimal.
2. The table also has a Date type column.
3. Create a question and add it to a dashboard.
4. Download the results to CSV.

Partial results from test to show difference in number format:

> Car Model,Price ($),Color,Date Sold
> Toyota Camry,"20,000",Red,"January 1, 2023"
> Honda Accord,"25,000",Blue,"January 2, 2023"
> Ford F-150,"30,000",Green,"January 3, 2023"
> 
> 1989 Ford Bronco,999,Red,"February 7, 2023"
> 1993 Mazda 626,650,White,"February 8, 2023"
> 1986 Toyota Pickup,800,Black,"February 9, 2023"
> 1992 Subaru Legacy,450,Green,"February 10, 2023"

<br>
<br>

 Partial results from test to show difference in date format:
> Car Model,Price ($),Color,Date Sold
> Toyota Camry,"20,000",Red,"January 1, 2023"
> Honda Accord,"25,000",Blue,"January 2, 2023"
> Ford F-150,"30,000",Green,"January 3, 2023"
> 
> Car Model,Price,Color,Date Sold
> Toyota Camry,20000,Red,2023-01-01
> Honda Accord,25000,Blue,2023-01-02
> Ford F-150,30000,Green,2023-01-03

### Expected behavior

1. Expecting consistency regardless of the length of the number, and no quotes around numbers on the export.
2. Date format to return to how it worked prior to 1.49

### Information about your Metabase installation

```JSON
My tests:
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Linux",
    "os.version": "5.15.49-linuxkit-pr",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "15.1 (Debian 15.1-1.pgdg110+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-03-14",
      "tag": "v1.49.0",
      "hash": "46c668b"
    },
    "settings": {
      "report-timezone": null
    }
  }
}

-- -- -- -- -- -- --

Customer:
{\rtf1\ansi\ansicpg1252\cocoartf2709
\cocoatextscaling0\cocoaplatform0{\fonttbl\f0\fswiss\fcharset0 Helvetica;}
{\colortbl;\red255\green255\blue255;\red59\green68\blue96;\red255\green255\blue255;}
{\*\expandedcolortbl;;\cssrgb\c29804\c34118\c45098;\cssrgb\c100000\c100000\c100000;}
\margl1440\margr1440\vieww11520\viewh8400\viewkind0
\deftab720
\pard\pardeftab720\partightenfactor0

\f0\fs28 \cf2 \cb3 \expnd0\expndtw0\kerning0
\outl0\strokewidth0 \strokec2 \{\
  "browser-info": \{\
    "language": "en-US",\
    "platform": "MacIntel",\
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36",\
    "vendor": "Google Inc."\
  \},\
  "system-info": \{\
    "file.encoding": "UTF-8",\
    "java.runtime.name": "OpenJDK Runtime Environment",\
    "java.runtime.version": "11.0.22+7",\
    "java.vendor": "Eclipse Adoptium",\
    "java.vendor.url": "https://adoptium.net/",\
    "java.version": "11.0.22",\
    "java.vm.name": "OpenJDK 64-Bit Server VM",\
    "java.vm.version": "11.0.22+7",\
    "os.name": "Linux",\
    "os.version": "5.10.210-201.852.amzn2.x86_64",\
    "user.language": "en",\
    "user.timezone": "GMT"\
  \},\
  "metabase-info": \{\
    "databases": [\
      "h2",\
      "mysql",\
      "athena"\
    ],\
    "hosting-env": "unknown",\
    "application-database": "mysql",\
    "application-database-details": \{\
      "database": \{\
        "name": "MySQL",\
        "version": "8.0.35"\
      \},\
      "jdbc-driver": \{\
        "name": "MariaDB Connector/J",\
        "version": "2.7.10"\
      \}\
    \},\
    "run-mode": "prod",\
    "version": \{\
      "date": "2024-03-21",\
      "tag": "v1.49.1",\
      "hash": "54ef5e9"\
    \},\
    "settings": \{\
      "report-timezone": null\
    \}\
  \}\
\}}
```

### Severity

P2 - Causing an issue for a customer who uses these exports to feed into other systems

**Comment by mkan-su** [comment]
Also the number having type % are downloaded with % symbol in version 49. For example CSV download has "50%" Vs 0.05 in versions before 49. This is causing problems wherever the csv is used. Issue present in open source version as well.

**Comment by adam-james-v** [comment]
@FilmonK , if the customer is able, or has already upgraded to a newer minor version (49.10 has format rows options/bug fixes), this should be solved. 

If they use the API to get downloads, there is a query param they can use `format_rows=false` to get downloads in the expected form.

If they are clicking the download button, they can hold 'Alt' and click the button to get an unformatted download. 
Finally, if they are using subscriptions, there is a checkbox 'use unformatted values in attachments' available.

If that is a workable solution for the customer, could you close this issue?

**Comment by adam-james-v** [comment]
@FilmonK , I'm closing this issue as it's fixed in 49.10 and later.
